src/Collapse/include/gudhi/Flag_complex_edge_collapser.h

```diff
@@ -14,367 +15,319 @@
+    auto edges2 = to_range<std::vector<typename Edge_collapser::Filtered_edge>>(std::forward<FilteredEdgeRange>(edges));
 #ifdef GUDHI_USE_TBB
-    tbb::parallel_sort(f_edge_vector_.begin(), f_edge_vector_.end(), sort_by_filtration);
+    // I think this sorting is always negligible compared to the collapse, but parallelizing it shouldn't hurt.
+    tbb::parallel_sort(edges2.begin(), edges2.end(),
+        [](auto const&a, auto const&b){return std::get<2>(a)>std::get<2>(b);});
 #else
-    std::sort(f_edge_vector_.begin(), f_edge_vector_.end(), sort_by_filtration);
+    std::sort(edges2.begin(), edges2.end(), [](auto const&a, auto const&b){return std::get<2>(a)>std::get<2>(b);});
 #endif
-
-    for (Edge_index endIdx = 0; endIdx < f_edge_vector_.size(); endIdx++) {
-      Filtered_edge fec = f_edge_vector_[endIdx];
-      Vertex_handle u = std::get<0>(fec);
-      Vertex_handle v = std::get<1>(fec);
-      Filtration_value filt = std::get<2>(fec);
-
-      // Inserts the edge in the sparse matrix to update the graph (G_i)
-      IEdge ie = insert_new_edge(u, v, endIdx);
-
-      iedge_to_index_map_.emplace(ie, endIdx);
-      critical_edge_indicator_.push_back(false);
-
-      if (!edge_is_dominated(u, v)) {
-        critical_edge_indicator_[endIdx] = true;
-        filtered_edge_output(u, v, filt);
-        if (endIdx > 1)
-          set_edge_critical(endIdx, filt, filtered_edge_output);
-      }
-    }
+    Edge_collapser edge_collapser;
+    edge_collapser.process_edges(edges2, std::forward<Delay>(delay));
+    return edge_collapser.output();
   }
-
-};
+  return std::vector<typename Edge_collapser::Filtered_edge>();
+}
 
 /** \brief Implicitly constructs a flag complex from edges as an input, collapses edges while preserving the persistent
- * homology and returns the remaining edges as a range.
+ * homology and returns the remaining edges as a range. The filtration value of vertices is irrelevant to this function.
  *
- * \param[in] edges Range of Filtered edges.There is no need the range to be sorted, as it will be performed.
+ * \param[in] edges Range of Filtered edges. There is no need for the range to be sorted, as it will be done internally.
  *
- * \tparam FilteredEdgeRange furnishes `std::begin` and `std::end` methods and returns an iterator on a
- * FilteredEdge of type `std::tuple<Vertex_handle, Vertex_handle, Filtration_value>` where `Vertex_handle` is the type
- * of a vertex index and `Filtration_value` is the type of an edge filtration value.
+ * \tparam FilteredEdgeRange Range of `std::tuple<Vertex_handle, Vertex_handle, Filtration_value>`
+ * where `Vertex_handle` is the type of a vertex index.
  *
  * \return Remaining edges after collapse as a range of
  * `std::tuple<Vertex_handle, Vertex_handle, Filtration_value>`.
- * 
+ *
  * \ingroup edge_collapse
- * 
+ *
+ * \note
+ * Advanced: Defining the macro GUDHI_COLLAPSE_USE_DENSE_ARRAY tells gudhi to allocate a square table of size the
```

Comment:
If you turn `GUDHI_COLLAPSE_USE_DENSE_ARRAY` into a template parameter, maybe it doesn't worth it, but a CMake variable to activate/deactivate this feature would be more usefull (C++ and python). Not mandatory on this PR.

Reply:
It really is a temporary advanced feature for people who know what they are doing, doing cmake stuff for it would be a waste of time. The real alternative is not documenting it at all.

Reply:
Ok, let's keep it as it is.